gobblin-core/src/main/java/gobblin/instrumented/Instrumented.java

```diff
@@ -131,6 +138,85 @@ public static MetricContext getMetricContext(State state, Class<?> klazz, List<T
   }
 
   /**
+   * Copies a metric context and returns a new context with different name, and possibly different tags.
+   * @param context Context to copy.
+   * @param tags New tags to apply to context. Repeated keys will override old tags.
+   * @param name Name of the new context. If absent or empty, will modify old name by adding a random integer at the end.
+   * @return Copied context.
+   */
+  public static MetricContext copyMetricContext(MetricContext context, List<Tag<?>> tags, Optional<String> name) {
+
+    String newName = name.orNull();
+
+    if (Strings.isNullOrEmpty(newName)) {
+      int randomId = RAND.nextInt();
+
+      String oldName = context.getName();
+      List<String> splitName = Splitter.on(".").splitToList(oldName);
+      if (StringUtils.isNumeric(Iterables.getLast(splitName))) {
```

Comment:
So you are pretty much assuming that if the last component is a number, then it is semantically the same as a random integer. I don't think this is a safe thing to do.